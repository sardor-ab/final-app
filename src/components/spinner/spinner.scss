// @import "src/style-utils";

.observer-spinner {
    user-select: none;
	display: inline-flex;
	position: relative;


    .circle {
        animation: pulse 1s ease-in-out infinite, view .2s ease-in-out backwards;
        animation-delay: 0s, 0s;
        animation-direction: normal, normal;
        animation-duration: 1s, .2s;
        animation-fill-mode: none, backwards;
        animation-iteration-count: infinite, 1;
        animation-play-state: running, running;
        animation-timing-function: ease-in-out, ease-in-out;
        background-color: green; //$mfr-green;
        border-radius: 6px;
        box-sizing: border-box;
        height: 12px;
        -webkit-transition: opacity .2s ease-out;
        transition: opacity .2s ease-out;
        width: 12px;

        &:nth-of-type(2) {
            animation-delay: .366s, .1s;
            margin: 0 8px;
        }

        &:nth-of-type(3) {
            animation-delay: .53s, .2s;
            background-color: lightcoral; //$mfr-lilac;
        }
    }

    @keyframes pulse {

        0% {
            transform: scale(1);
        }

        33% {
            transform: scale(1.3);
        }

        66% {
            transform: scale(1);
        }

        to {
            transform: scale(1);
        }
    }

    @keyframes view {

        from {
            opacity: 0;
            transform: scale(0);
        }

        to {
            opacity: 1;
            transform: scale(1);
        }
    }
}

