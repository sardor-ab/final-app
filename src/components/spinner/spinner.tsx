import React from 'react';
import cn from 'classnames';
import './spinner.scss';

type TProps = {
    className?: string;
};

export const Spinner = (props: TProps) => {
    const { className } = props;

    return (
        <div data-test-id="spinner" className={cn(className, 'observer-spinner')}>
            <div className="circle" />
            <div className="circle" />
            <div className="circle" />
        </div>
    );
};
