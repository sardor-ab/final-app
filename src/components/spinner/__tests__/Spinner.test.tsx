import React from 'react';
import { render } from '@testing-library/react';

import { Spinner } from 'components/spinner';

describe('Spinner test', () => {
    test('snapshot', () => {
        const { asFragment } = render(<Spinner />);

        expect(asFragment()).toMatchSnapshot();
    });
});
