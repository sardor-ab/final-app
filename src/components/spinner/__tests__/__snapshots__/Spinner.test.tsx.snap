// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner test snapshot 1`] = `
<DocumentFragment>
  <div
    class="observer-spinner"
    data-test-id="spinner"
  >
    <div
      class="circle"
    />
    <div
      class="circle"
    />
    <div
      class="circle"
    />
  </div>
</DocumentFragment>
`;
